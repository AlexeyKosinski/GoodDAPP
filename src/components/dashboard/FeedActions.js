// @flow
import React from 'react'
import { Alert, TouchableHighlight, View } from 'react-native-web'
import normalize from 'react-native-elements/src/helpers/normalizeText'
import { withStyles } from '../../lib/styles'
import { Icon, Section, Text } from '../common'
import type { FeedEventProps } from './FeedItems/EventProps'

/**
 * Returns swipeable actions for items inside Feed list
 *
 * @param {FeedEventProps} feedItem - Contains the feed item
 * @returns React element with actions
 */
const FeedActions = ({ item, styles, theme }: FeedEventProps) => (
  <Section.Row
    style={[
      styles.actionsContainer,
      { backgroundColor: item && item.type !== 'empty' ? theme.colors.red : theme.colors.surface },
    ]}
    alignItems="center"
    justifyContent="flex-end"
    grow
  >
    {item && item.type !== 'empty' && (
      <TouchableHighlight
        onPress={() => {
          Alert.alert('Tips', 'You could do something with this remove action!')
        }}
      >
        <View>
          <Section.Row justifyContent="center">
            <Icon name="close" color={theme.colors.surface} />
          </Section.Row>
          <Section.Row justifyContent="center">
            <Text color="surface">Delete</Text>
          </Section.Row>
        </View>
      </TouchableHighlight>
    )}
  </Section.Row>
)

const getStylesFromProps = ({ theme }) => ({
  actionsContainer: {
    borderRadius: normalize(8),
    height: normalize(84),
    maxHeight: normalize(84),
    padding: theme.sizes.default,
    marginHorizontal: theme.sizes.default,
  },
})

export default withStyles(getStylesFromProps)(FeedActions)
