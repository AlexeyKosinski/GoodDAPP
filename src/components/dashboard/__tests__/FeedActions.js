import React from 'react'
import renderer from 'react-test-renderer'
import { withThemeProvider } from '../../../__tests__/__util__'
import FeedActions from '../FeedActions'

// Note: test renderer must be required after react-native.

describe('FeedActions', () => {
  const WrappedFeedActions = withThemeProvider(FeedActions)
  it('renders without errors', () => {
    const tree = renderer.create(<WrappedFeedActions />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<WrappedFeedActions />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
