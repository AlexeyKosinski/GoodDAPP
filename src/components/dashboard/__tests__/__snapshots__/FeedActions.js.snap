// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedActions matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxFlex": 1,
        "WebkitBoxOrient": "horizontal",
        "WebkitBoxPack": "end",
        "WebkitFlexBasis": "0%",
        "WebkitFlexDirection": "row",
        "WebkitFlexGrow": 1,
        "WebkitFlexShrink": 1,
        "WebkitJustifyContent": "flex-end",
        "alignItems": "center",
        "backgroundColor": "rgba(255,255,255,1.00)",
        "borderBottomLeftRadius": "8px",
        "borderBottomRightRadius": "8px",
        "borderTopLeftRadius": "8px",
        "borderTopRightRadius": "8px",
        "flexBasis": "0%",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "84px",
        "justifyContent": "flex-end",
        "marginLeft": "8px",
        "marginRight": "8px",
        "maxHeight": "84px",
        "msFlexAlign": "center",
        "msFlexDirection": "row",
        "msFlexNegative": 1,
        "msFlexPack": "end",
        "msFlexPositive": 1,
        "msFlexPreferredSize": "0%",
        "paddingBottom": "8px",
        "paddingLeft": "8px",
        "paddingRight": "8px",
        "paddingTop": "8px",
      }
    }
  />
</div>
`;
